dspace-api/src/main/java/org/dspace/core/Utils.java

```diff
@@ -453,6 +456,40 @@ public static String getHostName(String uriString) {
         }
     }
 
+    /**
+     * Retrieve the IP address(es) of a given URI string.
+     * <P>
+     * At this time, DSpace only supports IPv4, so this method will only return IPv4 addresses.
+     * @param uriString URI string
+     * @return IP address(es) in a String array (or null if not found)
+     */
+    public static String[] getIPAddresses(String uriString) {
```

Comment:
We could consider to introduce a NetworkUtility injected via spring removing the use of the final class/static method. This would avoid the need to switch to mockito-inline.
This is just a suggestion, I'm not setting a block on that

Reply:
@abollini : While I agree in general we should move away from static methods & towards injected utilities, a `NetworkUtility` method in this scenario would *not* get rid of `mockito-inline` entirely.  If you look closely, the main reason for `mockito-inline` was to Mock the response from `InetAddress` (which we don't control): https://github.com/DSpace/DSpace/pull/3171/files#diff-3875fd7e7440313cffcd8ecdae3f1dd00d811ecc1fc641e58fab92380281d9b9R94

I guess it might be possible to wrap the InetAddress call in a method which we can then Mock, but unfortunately this is work that I'm worried won't get completed in time for Beta 5.  So, I'd rather we consider this one of the many many static methods we may want to replace.

(As a sidenote, it's also worth noting we already were using an earlier version of mockito-inline in `dspace-services`.  There was an `org.mockito.plugins.MockMaker` file which included the contents `mock-maker-inline`....that file was the old way of enabling `mockito-inline`, as it used to not be packaged/released separately.  So, in this PR, we've just switched to using the new package.)